### ISSUE TITLE
macos source build failure "cd: .deps: No such file or directory"

### ISSUE THREAD

**Issue by craigmac** [neovim/neovim#16509]
### Neovim version (nvim -v)

0.6.0

### Vim (not Nvim) behaves the same?

no

### Operating system/version

macOS 10.15.7

### Terminal name/version

iTerm 3.4.14

### $TERM environment variable

xterm-256color

### Installation

build from repo

### How to reproduce the issue

```
mkdir -p ~/tmp/fizz && cd ~/tmp/fizz
git clone https://github.com/neovim/neovim
make CMAKE_BUILD_TYPE=RelWithDebInfo
 ```

Same error if we checkout 'stable' instead and try the same, with a fresh rm -rf of the repo.

Again tried using the wiki build from source instructions instead using:

```
git clone https://github.com/neovim/neovim
cd neovim 
make -j4
```

### Expected behavior

Builds from source successfully.

### Actual behavior

Same errors using all three methods, which is:

```
cd .deps && \
		cmake -G 'Ninja'   \
		 /Users/cmaceach/git/neovim//third-party
/bin/sh: line 0: cd: .deps: No such file or directory
make: *** [build/.ran-third-party-cmake] Error 1
```

It looks like the directory is expanding to include an additional '/' for some reason? The `.deps` file is there and I can `cd` into it.

cmake --version:  3.19.8
ninja --version: 1.10.2

**Comment by craigmac** [comment]
I should add that I have been able to successfully build from source previously on tag 0.5.1 and other tags before that.

**Comment by clason** [comment]
If you have `ninja` installed, you should _not_ use `make -j` -- this will try to build dependencies in parallel, which fails.

Do `make distclean; make` -- I compile from source on macOS all the time without issues.

**Comment by craigmac** [comment]
Just ran `make distclean; make` and got the same Error, anything else that I could try?

**Comment by clason** [comment]
Is that with the wiki build instructions (for macOS, _not_ the quick start!)? Can you do this with a fresh clone and describe the exact steps (and put the _complete_ build log in a gist and share the link)?

**Comment by clason** [comment]
Oh, and what's your shell?

**Comment by craigmac** [comment]
Using instructions from README.md at https://github.com/neovim/neovim#install-from-source:

```sh
cd ~/git
rm -rf neovim && git clone https://github.com/neovim/neovim
cd neovim
make CMAKE_BUILD_TYPE=RelWithDebInfo
```

Output:

```
mkdir -p ".deps"
cd .deps && \
		cmake -G 'Ninja'   \
		 /Users/cmaceach/git/neovim//third-party
/bin/sh: line 0: cd: .deps: No such file or directory
make: *** [build/.ran-third-party-cmake] Error 1
```

Using instructions from here (macOS specific, *not* Quick Start at top): https://github.com/neovim/neovim/wiki/Building-Neovim#macos

```
cd ~/git && rm -rf neovim/
git clone https://github.com/neovim/neovim && cd neovim
make
```

Output:

```
mkdir -p ".deps"
cd .deps && \
		cmake -G 'Ninja'   \
		 /Users/cmaceach/git/neovim//third-party
/bin/sh: line 0: cd: .deps: No such file or directory
make: *** [build/.ran-third-party-cmake] Error 1
```

FWIW the `brew install neovim` method builds successfully and I can use Neovim 0.6.0 that way if I can't figure this out, but ideally I'd like to do the git way and keep building from source.

Shell is zsh, `echo $0` gives `-zsh`. No symlink on `ll \`which zsh\``.

**Comment by clason** [comment]
> Shell is zsh, echo $0 gives -zsh. 

For me, this gives `zsh` (no dash) when running zsh (and those instructions build fine, despite the double slash). Not sure what I can tell you further...

**Comment by craigmac** [comment]
`zsh` means it was run as a non-login session. I can emulate that by just starting a subshell by doing `zsh` and then when I do `echo $0` is get the same as yours `zsh` (minus the leading '-'). I tried the same build steps in this type of zsh, same result.

**Comment by clason** [comment]
Yeah, that was a stab in the dark. (I meant symlinks in the path to the neovim repo, by the way.)

**Comment by craigmac** [comment]
Nope, two folders `~/git` is regular folder with `755` and `neovim` inside the same.

**Comment by craigmac** [comment]
Tried building `0.5.1` tag again from scratch and got the same error, so something has changed in my env since the last successful build. I ended up bisecting my PATH value until it worked. Something I added to my PATH since last build was doing it. I successfully build 0.5.1 again with just `make distclean; make`, after this I was able to successfully build 'stable' with the same. I believe it was some newly added Android SDK stuff to my PATH that was being found first and causing issues. Thanks for all your help!
